lib/pages/chat/events/message/message.dart

```diff
@@ -708,6 +739,84 @@ class Message extends StatelessWidget {
 
     return currentEvent.senderId != comparedEvent.senderId;
   }
+
+  TextPainter _getSizeDisplayName(
+    BuildContext context,
+    double maxWidth,
+  ) {
+    return TextPainter(
+      text: TextSpan(
+        text: event.senderFromMemoryOrFallback
+            .calcDisplayname()
+            .shortenDisplayName(
+              maxCharacters: maxCharactersDisplayName,
+            ),
+        style: Theme.of(context).textTheme.labelMedium?.copyWith(
+              color: Theme.of(
+                context,
+              ).colorScheme.primary,
+            ),
+      ),
+      maxLines: 2,
+      textDirection: TextDirection.ltr,
+    )..layout(minWidth: 0, maxWidth: maxWidth);
+  }
+
+  TextPainter _getSizeMessageText(
+    BuildContext context,
+    double maxWidth,
+  ) {
+    return TextPainter(
+      text: TextSpan(
+        text: event.calcLocalizedBodyFallback(
+          MatrixLocals(L10n.of(context)!),
+          hideReply: true,
+        ),
+        style: Theme.of(context).textTheme.labelMedium?.copyWith(
+              color: Theme.of(
+                context,
+              ).colorScheme.primary,
+            ),
+      ),
+      textDirection: TextDirection.ltr,
+    )..layout(minWidth: 0, maxWidth: maxWidth);
+  }
+
+  double? _getSizeMessageBubbleWidth(
+    BuildContext context, {
+    required double maxWidth,
+    bool ownMessage = false,
+    bool hideDisplayName = false,
+  }) {
+    final isNotSupportCalcSize = {
+      MessageTypes.File,
+      MessageTypes.Image,
+      MessageTypes.Video,
+    }.contains(event.messageType);
+
+    if (ownMessage || hideDisplayName) {
+      return null;
+    }
+
+    if (isNotSupportCalcSize) {
+      return null;
+    }
+    final sizeWidthDisplayName = _getSizeDisplayName(
+      context,
+      maxWidth,
+    ).width;
+
+    final sizeWidthMessageText = _getSizeMessageText(
+      context,
+      maxWidth,
+    ).width;
+
+    if (sizeWidthDisplayName > sizeWidthMessageText) {
+      return sizeWidthDisplayName;
+    } else {
+      return sizeWidthMessageText;
+    }
+  }
```

Comment:
you could do smth fancy like this 
```suggestion
return max<int>(sizeWidthDisplayName, sizeWidthMessageText);
```